plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -132,14 +136,16 @@ public Map<String, Map<String, HiveColumnStatistics>> getPartitionColumnStatisti
             Set<String> columnNames)
     {
         Map<String, List<CompletableFuture<GetColumnStatisticsForPartitionResult>>> resultsForPartition = new HashMap<>();
+        HiveTableName hiveTableName = hiveTableName(databaseName, tableName);
         for (String partitionName : partitionNames) {
+            HivePartitionName hivePartitionName = hivePartitionName(hiveTableName, partitionName);
             ImmutableList.Builder<CompletableFuture<GetColumnStatisticsForPartitionResult>> futures = ImmutableList.builder();
             for (List<String> columnBatch : Lists.partition(ImmutableList.copyOf(columnNames), GLUE_COLUMN_READ_STAT_PAGE_SIZE)) {
                 GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
                         .withDatabaseName(databaseName)
                         .withTableName(tableName)
                         .withColumnNames(columnBatch)
-                        .withPartitionValues(partitionName);
+                        .withPartitionValues(hivePartitionName.getPartitionValues());
```

Comment:
we don't need `HivePartitionName` object.
we could just call `HivePartitionName.toPartitionValues(partitionName)` here, right?